java-type-checker/java_type_checker/expressions.py

```diff
@@ -88,6 +109,30 @@ def __init__(self, receiver, method_name, *args):
         self.method_name = method_name
         self.args = args
 
+    def static_type(self):
+        return self.receiver.static_type().method_named(self.method_name).return_type
+
+    def check_types(self):
+        self.receiver.check_types()
+        for arg in self.args:
+            arg.check_types()
+
+        self.receiver.static_type().method_named(self.method_name)
+        parameter_types = self.receiver.static_type().method_named(self.method_name).parameter_types
+        if len(self.args) != len(parameter_types):
+            raise JavaArgumentCountError("Wrong number of arguments for " + self.receiver.static_type().name + "." + self.method_name + "(): expected " + str(len(
+                parameter_types)) + ", got " + str(len(self.args)))
+
+        args_types = []
+        raise_error = False
+        for i in range(len(self.args)):
+            args_types.append(self.args[i].static_type().name)
+            if not self.args[i].static_type().is_subtype_of(parameter_types[i]):
+                raise_error = True
```

Comment:
I made this weird solution so I could find the list of all type names, but I totally missed that there was a _names() function that does it for me.